packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -196,29 +199,153 @@ class CupertinoAlertDialog extends StatelessWidget {
       children.add(const Padding(padding: const EdgeInsets.only(top: 8.0)));
     }
 
+    return new ClipRRect(
+      borderRadius: const BorderRadius.only(
+        topLeft: const Radius.circular(12.0),
+        topRight: const Radius.circular(12.0),
+      ),
+      child: new Container(
+        color: _kDialogForegroundColor,
+        child: new Column(
+          mainAxisSize: MainAxisSize.min,
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          children: children,
+        ),
+      ),
+    );
+  }
+
+  Widget _buildActions() {
+    Widget actionSection = new Container();
     if (actions.isNotEmpty) {
-      final Widget actionSection = new _CupertinoAlertActionSection(
+      actionSection = new _CupertinoAlertActionSection(
         children: actions,
         scrollController: actionScrollController,
       );
-      children.add(
-        new Flexible(child: actionSection),
-      );
     }
 
-    return new Padding(
-      padding: const EdgeInsets.symmetric(vertical: _kEdgePadding),
-      child: new CupertinoDialog(
-        child: new Column(
-          mainAxisSize: MainAxisSize.min,
-          crossAxisAlignment: CrossAxisAlignment.stretch,
-          children: children,
+    return new ClipRRect(
+      borderRadius: const BorderRadius.only(
+        bottomLeft: const Radius.circular(12.0),
+        bottomRight: const Radius.circular(12.0),
+      ),
+      child: actionSection,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Center(
+      child: new Container(
+        width: _kCupertinoDialogWidth,
+        padding: const EdgeInsets.symmetric(vertical: _kEdgePadding),
+        child: new CustomMultiChildLayout(
+          delegate: new _CupertinoAlertDialogLayoutDelegate(
+            actionsCount: actions.length,
+          ),
+          children: <Widget>[
+            new LayoutId(
+              id: _AlertDialogSection.blur,
+              child: _buildBlurBackground(),
+            ),
+            new LayoutId(
+              id: _AlertDialogSection.content,
+              child: _buildContent(),
+            ),
+            new LayoutId(
+              id: _AlertDialogSection.actions,
+              child: _buildActions(),
+            ),
+          ],
         ),
       ),
     );
   }
 }
 
+enum _AlertDialogSection {
+  blur,
+  content,
+  actions,
+}
+
+class _CupertinoAlertDialogLayoutDelegate extends MultiChildLayoutDelegate {
+
+  final int actionsCount;
+
+  _CupertinoAlertDialogLayoutDelegate({
+    this.actionsCount,
+  });
+
+  @override
+  void performLayout(Size size) {
+    // TODO: @mattcarroll, handle 2 buttons that need to stack due to long text
```

Comment:
I don't really have an opinion but the rest of the todos have the format `// TODO(mattcarroll):`. Worth keeping with convention.

Reply:
the exact syntax is in the style guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo